pkg/kubectl/cmd/create_job.go

```diff
@@ -62,86 +58,89 @@ func NewCmdCreateJob(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
 		Out: cmdOut,
 	}
 	cmd := &cobra.Command{
-		Use:     "job --from-cronjob=CRONJOB",
+		Use:     "job NAME [--from-cronjob=CRONJOB]",
 		Short:   jobLong,
 		Long:    jobLong,
 		Example: jobExample,
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(c.Complete(f, cmd, args))
-			cmdutil.CheckErr(c.RunCreateJob())
+			cmdutil.CheckErr(c.RunCreateJob(f))
 		},
 	}
 	cmdutil.AddApplyAnnotationFlags(cmd)
 	cmdutil.AddValidateFlags(cmd)
 	cmdutil.AddPrinterFlags(cmd)
-	cmd.Flags().String("from-cronjob", "", "Specify the name of the CronJob to create a Job from.")
+	cmd.Flags().String("from", "", "The name of the resource to create a Job from (only cronjob is supported).")
 
 	return cmd
 }
 
 func (c *CreateJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) (err error) {
-	c.FromCronJob = cmdutil.GetFlagString(cmd, "from-cronjob")
-
-	// Complete other options for Run.
-	c.Mapper, _ = f.Object()
-
-	c.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+	if len(args) == 0 {
+		return cmdutil.UsageErrorf(cmd, "NAME is required")
+	}
+	c.Name = args[0]
 
+	c.From = cmdutil.GetFlagString(cmd, "from")
 	c.Namespace, _, err = f.DefaultNamespace()
 	if err != nil {
 		return err
 	}
 
 	c.PrintObject = func(obj runtime.Object) error {
-		return f.PrintObject(cmd, obj, c.Out)
+		return cmdutil.PrintObject(cmd, obj, c.Out)
 	}
 
-	// need two client sets to deal with the differing versions of CronJobs and Jobs
 	clientset, err := f.KubernetesClientSet()
 	if err != nil {
 		return err
 	}
-	if c.V1Client == nil {
-		c.V1Client = clientset.BatchV1()
-	}
-	if c.V1Beta1Client == nil {
-		c.V1Beta1Client = clientset.BatchV1beta1()
-	}
+	c.Client = clientset.BatchV1()
 
 	return nil
 }
 
-func (c *CreateJobOptions) RunCreateJob() (err error) {
-	cronjob, err := c.V1Beta1Client.CronJobs(c.Namespace).Get(c.FromCronJob, metav1.GetOptions{})
-
+func (c *CreateJobOptions) RunCreateJob(f cmdutil.Factory) error {
```

Comment:
We really need to avoid passing the factory.  Some places pass a builder around in their struct which I can live with.